internal/core/graphics/color.rs

```diff
@@ -192,6 +194,52 @@ impl Color {
         let rgba: RgbaColor<f32> = hsva.into();
         rgba.into()
     }
+
+    /// Returns a new version of this color with the opacity changed by _subtracting_
+    /// `amount` from it. The reference is the opacity's normalized value as `f32`.
+    ///
+    /// This means that a _positive_ `amount` will yield a more transparent color and
+    /// a _negative_ `amount` will yield a more opaque color.
+    #[must_use]
+    pub fn faded(&self, amount: f32) -> Self {
+        let mut rgba: RgbaColor<f32> = (*self).into();
+        rgba.alpha = rgba.alpha.sub(amount).clamp(0.0, 1.0);
+        rgba.into()
+    }
+
+    /// Returns a new color that is a mix of `self` and `other`, with a proportion
+    /// factor given by `factor` (which should be between `0.0` and `1.0`).
+    ///
+    /// This will make a weighted average of the corresponding RGBA channels,
+    /// with weight `factor` for `self` and `1 - factor` for `other`, and produce
+    /// the color with the resulting values.
+    #[must_use]
+    pub fn mixed(&self, other: &Self, factor: f32) -> Self {
+        // * NOTE: The implemented behavior is based on the `mix` function from SASS
+
+        let s: RgbaColor<u8> = (*self).into();
+        let o: RgbaColor<u8> = (*other).into();
+        let factor = factor.clamp(0.0, 1.0);
+
+        // cSpell: ignore wavg
+        let wavg = |a: u8, b: u8| -> u8 { (a as f32 * factor + b as f32 * factor).round() as u8 };
```

Comment:
What about using `lerp` instead of `wavg`?

I mean, `lerp` is exactly what I'm doing, and `wavg` (for _weighted average_) is more general than this...